Packs/Workday/Integrations/WorkdayIAM/WorkdayIAM.py

```diff
@@ -0,0 +1,294 @@
+                }
+                events.append(event)
+            last_run_time = datetime.now().strftime(WORKDAY_DATE_TIME_FORMAT)
+            demisto.info(f'Workday Fetch Events Completed. Response Time:'
+                         f' {(datetime.now() - start).total_seconds()} seconds')
+
+        last_run = {'time': last_run_time, "sync_users": True}
+    except Exception as e:
+        demisto.error(f'Failed to fetch events. From Date = {from_date_time}. To Date = {to_date_time}')
+        raise e
+
+    return last_run, events
+
+
+def get_demisto_user(workday_user):
+    employee_id = workday_user.get('employeeid')
+    data = demisto.searchIndicators(query=f"type:\"User Profile\" and employeeid:\"{employee_id}\"").get('iocs')
+    if data:
+        return data[0]
+    return None
+
+
+def does_email_exist_in_xsoar(email_address):
+    data = demisto.searchIndicators(query=f"type:\"User Profile\" and value:\"{email_address}\"").get('iocs')
+    does_email_exist = (data and len(data) > 0)
+    return does_email_exist
+
+
+def get_profile_changed_fields(workday_user, demisto_user):
+    if not demisto_user:
+        return False  # potential new hire
+    profile_changed_fields = []
+    for user_profile_key in workday_user.keys():
+        workday_value = workday_user.get(user_profile_key)
+        demisto_value = demisto_user.get('CustomFields', {}).get(user_profile_key)
+        if workday_value and demisto_value and workday_value != demisto_value:
+            profile_changed_fields.append(user_profile_key)
+
+    return profile_changed_fields
+
+
+def check_if_user_should_be_terminated(workday_user):
+    # check if employee is active and his terminate fay or last day of work arrived
+    is_term_event = False
+    employment_status = str(workday_user.get(EMPLOYMENT_STATUS_EVENT_FIELD))
+
+    if employment_status.lower() == EMPLOYEE_ACTIVE_STATUS:
+        last_day_of_work = workday_user.get(LAST_DAY_OF_WORK_EVENT_FIELD)
+        termination_date = workday_user.get(TERMINATION_DATE_EVENT_FIELD)
+        if last_day_of_work or termination_date:
+            hire_date = workday_user.get(HIRE_DATE_EVENT_FIELD)
+            hire_date = datetime.strptime(hire_date, WORKDAY_DATE_FORMAT)
+            today = datetime.today()
+            # Check if term date is older than the latest hire date. If it is, then it is not a term, its a rehire
+            # Also check with current date. If its future date, then it's not a term event
+            if last_day_of_work:
+                last_day_of_work = datetime.strptime(last_day_of_work, WORKDAY_DATE_FORMAT)
+                is_term_event = hire_date <= last_day_of_work <= today
+            elif termination_date:
+                termination_date = datetime.strptime(termination_date, WORKDAY_DATE_FORMAT)
+                is_term_event = hire_date <= termination_date <= today
+    return is_term_event
+
+
+def test_module(client, params, mapper_in):
```

Comment:
```suggestion
def test_module(client, is_fetch, fetch_events_time_minutes , mapper_in):
```
pass each param instead of all the params